File: Seccion 11 - Funciones en C++/Seccion_11_Dart/15_ejercicio_12.dart

import 'dart:io';

class Vector {
  int tam;
  List<int> valores = new List();

  Vector();
}

void pedirDatos(Vector vector) {
  stdout.writeln('Digite el número de elementos del vector: ');
  vector.tam = int.parse(stdin.readLineSync());

  for (int i = 0; i < vector.tam; i++) {
    stdout.writeln('${i + 1}. Digite un número: ');
    int valor = int.parse(stdin.readLineSync());
    vector.valores.insert(i, valor);
  }
}

void comprobarOrdenamiento(Vector vector) {
  bool band = false;

  int i = 0;
  while ((band == false) && (i < vector.tam - 1)) {
    if (vector.valores[i] > vector.valores[i + 1]) {
      band = true;
    }

    i++;
  }

  if (band == false) {
    print('El arreglo está ordenado crecientemente!');
  } else {
    print('El arreglo no está ordenado!');
  }
}

void main() {

  Vector vector = new Vector();

  pedirDatos(vector);

  comprobarOrdenamiento(vector);
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/15_ejercicio_12.py

tam = 0
vector = []

def pedir_datos():
    global vector, tam
    
    tam = int(input('Digite el número de elementos del vector: '))
    
    for i in range(tam):
        valor = int(input('{}. Digite un número: '.format(i + 1)))
        vector.append(valor)

def comprobar_ordenamiento():
    band = False
    
    i = 0
    while band == False and i < tam - 1:
        if vector[i] > vector[i + 1]:
            band = True
        
        i += 1
    
    if band == False:
        print('El arreglo está ordenado crecientemente!')
    else:
        print('El arreglo no está ordenado!')

pedir_datos()

comprobar_ordenamiento()